package/MDAnalysis/core/groups.py

```diff
@@ -640,16 +648,57 @@ def center(self, weights, pbc=None):
         else:
             xyz = atoms.positions
 
-        return np.average(xyz, weights=weights, axis=0)
-
-    def center_of_geometry(self, pbc=None):
+        if compound.lower() == 'group':
+            return np.average(xyz, weights=weights, axis=0)
+        elif compound.lower() == 'residues':
+            compound_indices = atoms.resindices
+            n_compounds = atoms.n_residues
+        elif compound.lower() == 'segments':
+            compound_indices = atoms.segindices
+            n_compounds = atoms.n_segments
+        else:
+            raise ValueError("Unrecognized compound definition: {}\nPlease use"
+                             " one of 'group', 'residues', or 'segments'."
+                             "".format(compound))
+
+        # Sort positions and masses by compound index:
+        sort_indices = np.argsort(compound_indices)
+        compound_indices = compound_indices[sort_indices]
+        positions = atoms.positions[sort_indices]
+        weights = weights[sort_indices]
+        # Allocate output array:
+        coms = np.zeros((n_compounds, 3), dtype=np.float32)
```

Comment:
should be named differently here. `centers` might be more appropriate than `coms`